### ISSUE TITLE
Passcodes Created via API Don't Always Trigger Webhook, Despite "100 success" Response

### ISSUE THREAD

**Issue by mynameissue** [OpenWonderLabs/SwitchBotAPI#345]
### Current Situation

I create passcodes for the keypad touch via API, but sometimes it doesn’t work properly.There are three possible cases:

1  The passcode is created correctly, and the webhook comes after a minute.
2  A "timeout" webhook comes after a minute.
3  Nothing happens; no webhook is received, and the passcode is not created, even after waiting for 3 hours.
In all cases, the API response is “100 success.”

### Logs

```markdown
{"statusCode":100,"body":{"commandId":"CMDxxxxxxxxxxxxxx"},"message":"success"}
```

### Configuration

```json
No external configuration file used. All settings are managed within the Google Apps Script code.
```

### Environment

-Platform: Google Apps Script (GAS)
-Runtime Version: V8
-Execution Environment: Google Cloud infrastructure (no local OS version or node/npm applicable)

**Comment by dds82** [comment]
I believe the issue I reported and closed (#345) is the same issue. I am still seeing the API commands to create and delete keys behave inconsistently.

**Comment by Larsene** [comment]
You're right, the API always send a success response. But this is the expected response for the Keypad, as explained in the documentation : https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#keypad-touch-2.

BUT, there's no webhook response too (as explained here : https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#keypad-3), and this is anormal.

Already reported, but SwitchBot doesn't read our reports and fix his API, this is really a pity...

**Comment by hsakoh** [comment]
Data point (AP region, 2026-06-22): Today I received webhooks for both createKey and deleteKey on a Keypad Touch (WoKeypadTouch) via the SwitchBot cloud webhook.

createKey:
```
{"eventType":"changeReport","eventVersion":"1","context":{"battery":84,"deviceMac":"<redacted>","deviceType":"WoKeypadTouch","eventName":"createKey","keyName":"test","result":"success","timeOfSample":1782113009472}}
```
deleteKey:
```
{"eventType":"changeReport","eventVersion":"1","context":{"battery":84,"deviceMac":"<redacted>","deviceType":"WoKeypadTouch","eventName":"deleteKey","keyName":"test","result":"success","timeOfSample":1782113782075}}
```
Note: the payload includes a keyName field that isn't in the official docs.

**Comment by Bu66as** [comment]
**Data point (EU / Germany region) — createKey/deleteKey webhooks not delivered**

Adding another affected environment to give this more weight:

- Devices: Smart Lock Ultra (FW V2.6), Keypad Touch (FW V2.5), Hub Mini 2 (Matter)
- Region: Europe (Germany)
- `createKey`/`deleteKey` via the cloud API return `{statusCode:100,"success"}`, and the key is correctly created on the keypad — but **no `changeReport` webhook ever arrives** for the keypad.
- `queryWebhook` shows our endpoint correctly registered, and the endpoint serves a valid (non-self-signed) TLS cert. We **do** receive webhooks for other devices (e.g. the lock's `lockState`), so the delivery path itself is fine — it's keypad events specifically that never fire.
- Lock webhooks are also unreliable here (`LOCKED` arrives, `UNLOCKED` often does not).

@Minsheng is there any update on this? It's been acknowledged for a while and several users across regions are blocked by it. Thanks for looking into it.
